src/modules/gatsby-transform-url/breakpoints.ts

```diff
@@ -0,0 +1,76 @@
+import ImgixClient from 'imgix-core-js';
+
+const VARIABLE_QUALITIES = [75, 50, 35, 23, 20];
+const MAX_SIZE = 8192;
+const MAX_DPR = 4;
+
+const clamp = (clamp: number, val?: number) => Math.min(val ?? clamp, clamp);
+const min = (fallback: number, ...rest: (number | undefined)[]) =>
+  Math.min(fallback, ...rest.filter<number>(isNotNull));
+
+const isNotNull = <T>(v: T): v is Exclude<T, undefined | null> => v != null;
+export const generateBreakpoints = (
+  opts: (
+    | {
+        layout: 'fullWidth';
+        width?: number;
+      }
+    | {
+        layout: 'fixed' | 'constrained';
+        width: number;
+      }
+  ) & {
```

Comment:
Are these going to be exposed to end users?

Reply:
This will be used inside our hook to generate the `breakpoints` field that is passed to `getImageData`, so it won't be exposed to users directly.